core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -149,10 +149,31 @@ public boolean queryStringLenient() {
         return this.queryStringLenient;
     }
 
-    IndicesQueriesRegistry indicesQueriesRegistry() {
+    public IndicesQueriesRegistry indicesQueriesRegistry() {
         return indicesQueriesRegistry;
     }
 
+    public ParsedQuery parse(QueryBuilder<?> queryBuilder) {
+        QueryShardContext context = cache.get();
+        context.reset();
+        context.parseFieldMatcher(parseFieldMatcher);
+        try {
+            return innerParse(context, queryBuilder);
+        } catch (ParsingException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new QueryShardException(context, "failed to create query: {}", e, queryBuilder);
+        }
+    }
+
+    private static ParsedQuery innerParse(QueryShardContext context, QueryBuilder<?> queryBuilder) throws IOException, QueryShardException {
```

Comment:
the adjustments have been made, no changes should be required in this class anymore as part of this PR